frontend/tests/controllers-test/authCtrl.test.js

```diff
@@ -191,23 +192,111 @@ describe('Unit Tests for auth controller', function() {
             vm.userLogin(true);
             var token = "encrypted";
             var response = utilities.sendRequest(vm.getUser);
-            expect(response.status).to.equal(200);
-            expect(response.data).to.equal("success");
-            expect(angular.equals(response.userKey, token)).to.equal(true);
+            expect(response.status).toEqual(200);
+            expect(response.data).toEqual("success");
+            expect(angular.equals(response.userKey, token)).toEqual(true);
         });
 
         it('backend error', function() {
             vm.getUser.username = '';
             var token = "notFound";
             var response = utilities.sendRequest(vm.getUser);
-            expect(response.status).to.equal(400);
-            expect(response.data).to.equal("error");
-            expect(angular.equals(response.userKey, token)).to.equal(true);
+            expect(response.status).toEqual(400);
+            expect(response.data).toEqual("error");
+            expect(angular.equals(response.userKey, token)).toEqual(true);
         });
 
         it('invalid details', function() {
             vm.userLogin(false);
-            expect($rootScope.isLoader).to.equal(false);
+            expect($rootScope.isLoader).toEqual(false);
+        });
+    });
+
+    describe('checkStrength', function() {
+
+        beforeEach(function() {
+            utilities.passwordStrength = function(password) {
+                //Regular Expressions.  
+                var regex = new Array();
+                regex.push("[A-Z]","[a-z]","[0-9]","[$$!%*#?&]");
+
+                var passed = 0;
+                //Validate for each Regular Expression.  
+                for (var i = 0; i < regex.length; i++) {
+                    if (new RegExp(regex[i]).test(password)) {
+                        passed++;
+                    }
+                }
+                //Validate for length of Password.  
+                if (passed > 2 && password.length > 8) {
+                    passed++;
+                }
+     
+                var color = "";
+                var strength = "";
+                if (passed == 1) {
+                    strength = "Weak";
+                    color = "red";
+                } else if (passed == 2) {
+                    strength = "Average";
+                    color = "darkorange";
+                } else if (passed == 3) {
+                    strength = "Good";
+                    color = "green";
+                } else if (passed == 4) {
+                    strength = "Strong";
+                    color = "darkgreen";
+                } else if (passed == 5) {
+                    strength = "Very Strong";
+                    color = "darkgreen";
+                }
+                return [strength, color];
+            }
+        });
+
+        it('check Weak password', function() {
```

Comment:
Club all the password strengths into one unit test, and maintain a list of objects with password, expectedStrength & expectedColor. Iterate over that list and do a similar type of validation inside for loop. Let's avoid duplication here, In this way adding more password strength checks will be much easier in future.

Reply:
@galipremsagar done. Please have a look if it's in the same way as you wanted.

Reply:
Hey, I meant something different, I've made those changes. You will find a patch file in this zip. Please apply this path with this entire file and push the changes.

[Merged_all_code_duplication_into_one_single_function_which_does_the_same___Co-authored-by_.patch.zip](https://github.com/Cloud-CV/EvalAI/files/3114908/Merged_all_code_duplication_into_one_single_function_which_does_the_same___Co-authored-by_.patch.zip)

Reply:
done.